src/accelerate/accelerator.py

```diff
@@ -2579,7 +2581,14 @@ def save_model(
         os.makedirs(save_directory, exist_ok=True)
 
```

Comment:
Can you move L2578-L2580 (checking if any param is on meta device) in the else block ?

Reply:
Oops I missed that, thanks for catching! Moved to the else block before the `self.get_state_dict` call